src/unaligned_vector/binary_quantized.rs

```diff
@@ -0,0 +1,299 @@
+                0b_00000000_00000000_00000000_00000001,
+                0b_00000000_00000000_00000000_00000010,
+                0b_00000000_00000000_00000000_00000100,
+                0b_00000000_00000000_00000000_00001000,
+            ];
+            let mask = vld1q_u32(mask.as_ptr());
+            let lane = vandq_u32(lane, mask);
+
+            let left = vaddvq_u32(lane) as u8;
+
+            let lane = vld1q_f32(ptr.add(i * 8 + 4));
+            let lane = vcltzq_f32(lane);
+            let lane = vmvnq_u32(lane);
+            let mask: Vec<u32> = vec![
+                0b_00000000_00000000_00000000_00010000,
+                0b_00000000_00000000_00000000_00100000,
+                0b_00000000_00000000_00000000_01000000,
+                0b_00000000_00000000_00000000_10000000,
+            ];
+            let mask = vld1q_u32(mask.as_ptr());
+            let lane = vandq_u32(lane, mask);
+
+            let right = vaddvq_u32(lane) as u8;
+
+            ret[i] = left | right;
+        }
+    }
+
+    // Since we're iterating on bytes two by two.
+    // If we had a number of dimensions not dividible by 8 we may be
+    // missing some bits in the last byte.
+    let reminder = slice.len() % size_of::<QuantizedWord>();
+    if reminder != 0 {
+        let mut rem: u8 = 0;
+        for r in slice[slice.len() - reminder..].iter().rev() {
+            rem <<= 1;
+            let r = r.is_sign_positive();
+            rem |= r as u8;
+        }
+        ret[iterations] = rem;
+    }
+
+    ret
+}
+
+pub(super) fn to_vec_non_optimized(vec: &UnalignedVector<BinaryQuantized>) -> Vec<f32> {
+    vec.iter().collect()
+}
+
+#[cfg(any(target_arch = "aarch64", target_arch = "arm64ec"))]
+unsafe fn to_vec_neon(vec: &UnalignedVector<BinaryQuantized>) -> Vec<f32> {
+    use core::arch::aarch64::*;
+
+    let mut output: Vec<f32> = vec![0.0; vec.len()];
+    let output_ptr = output.as_mut_ptr();
+    let bytes = vec.as_bytes();
+    let low_mask = [0b_0000_0001, 0b_0000_0010, 0b_0000_0100, 0b_0000_1000];
+    let high_mask = [0b_0001_0000, 0b_0010_0000, 0b_0100_0000, 0b_1000_0000];
+    let ones = unsafe { vld1q_dup_f32(&1.0) };
+    let minus = unsafe { vld1q_dup_f32(&-1.0) };
+
+    for (current_byte, base) in bytes.iter().enumerate() {
+        unsafe {
+            let base = *base as u32;
+            let base = vld1q_dup_u32(&base);
+            for (i, mask) in [low_mask, high_mask].iter().enumerate() {
+                let mask = vld1q_u32(mask.as_ptr());
+                let mask = vandq_u32(base, mask);
+                // 0xffffffff if equal to zero and 0x00000000 otherwise
+                let mask = vceqzq_u32(mask);
+                let lane = vbslq_f32(mask, minus, ones);
+                let offset = output_ptr.add(current_byte * 8 + i * 4);
+                vst1q_f32(offset, lane);
+            }
+        }
+    }
+
+    output
+}
+
+#[cfg(not(any(target_arch = "aarch64", target_arch = "arm64ec")))]
```

Comment:
This is not the right feature clause. Use the right one. Same for the neon one.

Reply:
Fixed in https://github.com/meilisearch/arroy/pull/82/commits/777effb409fc0ab9f797f93ba8eded676b6bb0c8